substrate/tendermint/client/src/block_import.rs

```diff
@@ -0,0 +1,186 @@
+use std::{marker::PhantomData, sync::Arc, collections::HashMap};
+
+use async_trait::async_trait;
+
+use sp_api::BlockId;
+use sp_runtime::traits::{Header, Block};
+use sp_blockchain::{BlockStatus, HeaderBackend, Backend as BlockchainBackend};
+use sp_consensus::{Error, CacheKeyId, BlockOrigin, SelectChain};
+
+use sc_consensus::{BlockCheckParams, BlockImportParams, ImportResult, BlockImport, Verifier};
+
+use sc_client_api::{Backend, BlockBackend};
+
+use crate::{TendermintValidator, tendermint::TendermintImport};
+
+impl<T: TendermintValidator> TendermintImport<T> {
+  fn check_already_in_chain(&self, hash: <T::Block as Block>::Hash) -> bool {
+    let id = BlockId::Hash(hash);
+    // If it's in chain, with justifications, return it's already on chain
+    // If it's in chain, without justifications, continue the block import process to import its
+    // justifications
+    // This can be triggered if the validators add a block, without justifications, yet the p2p
+    // process then broadcasts it with its justifications
+    if (self.client.status(id).unwrap() == BlockStatus::InChain) &&
```

Comment:
As in `if ((`? No, Rust warns about un-needed parentheses. It's polite enough to allow wrapping individual conditions for those who don't trust the compilers order of operations though, even though they're technically unnecessary. I personally believe in wrapping every condition/OoO ambiguity, yet don't mind the if statement itself being open.

Reply:
Let's wrap the second condition then please, no need to wrap both.

Reply:
The second condition is a single term with no spaces/operators, and parentheses delineate whitespace/operators. Adding parentheses to it would also trigger Rust/clippy.